src/MySteam.cpp

```
@@ -12,7 +13,32 @@
 #include "json/yajlHelpers.h"
 
 MySteam::MySteam() {
+    std::string data_home_path;
+    if (getenv("XDG_DATA_HOME") != NULL) {
+        data_home_path = getenv("XDG_DATA_HOME");
+    } else {
+        data_home_path = getenv("HOME") + std::string("/.local/share");
+    }
 
+    if (file_exists(data_home_path + "/Steam/appcache/appinfo.vdf")) {
+        m_steam_install_dir = std::string(data_home_path + "/Steam");
+        return;
+    }
+
+    const std::string home_path = getenv("HOME");
+    if (file_exists(home_path + "/.steam/appcache/appinfo.vdf")) {
+        m_steam_install_dir = std::string(home_path + "/.steam");
+        return;
+    }
+    else if (file_exists(home_path + "/.steam/steam/appcache/appinfo.vdf")) {
+        m_steam_install_dir = std::string(home_path + "/.steam/steam");
+        return;
+    }
+    else {
+        std::cerr << "Unable to locate the steam directory. TODO: implement a folder picker here" << std::endl;
+        system("zenity --error --no-wrap --text=\"Unable to find your Steam installation directory.. Please report this on Github!\"");
```

this is going to introduce another dependency on SamRewritten.
Although it is already a dependency of Steam.

---

Yeah but it's okay if it's not installed anyways, it'll just fail silently as always

---

eh, we probably just want to use a GtkDialog directly. Anytime there's an error, just create a dialog on the fly with the desired text. This is fine for now.